src/mixins/createCachedRenderMixin.ts

```diff
@@ -121,9 +131,15 @@ const createCachedRenderMixin: CachedRenderFactory = createStateful
 					props[key] = cachedRender.listeners[key];
 				}
 				const classes: { [index: string]: boolean; } = {};
+				const widgetClasses: string[] = widgetClassesMap.get(cachedRender);
+
+				widgetClasses.forEach((c) => classes[c] = false);
+
 				if (cachedRender.classes) {
 					cachedRender.classes.forEach((c) => classes[c] = true);
+					widgetClassesMap.set(cachedRender, cachedRender.classes);
```

Comment:
I don't think that we need to track the history prior to `render()` - 1, as the classes have already been removed and therefore are not required in the `classes` object and the first example will behave as we expect.

I wrote a functional test to prove this given the scenario above that classes are added and  removed as expected, please have a look and let me know any thoughts.